coord/src/rmr/redise_parser/grammar.y

```diff
@@ -1,218 +0,0 @@
-/*
- * Copyright Redis Ltd. 2016 - present
- * Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
- * the Server Side Public License v1 (SSPLv1).
- */
-
-%token_type {Token}
-%name MRTopologyRequest_Parse
-
-%include {
-#include "rmr/common.h"
-#include "token.h"
-#include "grammar.h"
-#include "parser_ctx.h"
-#include "../cluster.h"
-#include "../node.h"
-#include "../endpoint.h"
-
-#include <stdlib.h>
-#include <stdio.h>
-#include <assert.h>
-
-#include "lexer.h"
-
-static void syntax_error(parseCtx *ctx, const char *fmt, ...);
-
-} // END %include
-
-%syntax_error {
-    syntax_error(ctx, "Syntax error at offset %d near '%.*s'", TOKEN.pos, TOKEN.len, TOKEN.s);
-}
-
-%default_type { char * }
-%default_destructor { rm_free($$); }
-%extra_argument { parseCtx *ctx }
-%type shard { RLShard }
-%destructor shard { MRClusterNode_Free(&$$.node); }
-%type endpoint { MREndpoint }
-%destructor endpoint { MREndpoint_Free(&$$); }
-%type topology { MRClusterTopology * }
-%destructor topology { MRClusterTopology_Free($$); }
-
-%type master {int}
-%destructor master {}
-%destructor cluster {}
-%type tcp_addr {Token}
-%destructor tcp_addr {}
-
-root ::= cluster topology(D). {
-    if (ctx->numSlots) {
-        if (ctx->numSlots > 0 && ctx->numSlots <= 16384) {
-            D->numSlots = ctx->numSlots;
-        } else {
-            // ERROR!
-            syntax_error(ctx, "Invalid number of slots %d", ctx->numSlots);
-            goto err;
-        }
-    }
-    // translate optional shard func from arguments to proper enum.
-    // We will only get it from newer versions of the cluster, so if we don't get it we assume
-    // CRC12 / 4096
-    D->hashFunc = MRHashFunc_None;
-    if (ctx->shardFunc) {
-        if (!strncasecmp(ctx->shardFunc, MRHASHFUNC_CRC12_STR, strlen(MRHASHFUNC_CRC12_STR))) {
-            D->hashFunc = MRHashFunc_CRC12;
-        } else if (!strncasecmp(ctx->shardFunc, MRHASHFUNC_CRC16_STR, strlen(MRHASHFUNC_CRC16_STR))) {
-            D->hashFunc = MRHashFunc_CRC16;
-        } else {
-            // ERROR!
-            syntax_error(ctx, "Invalid hash func %s", ctx->shardFunc);
-            goto err;
-        }
-    }
-    ctx->topology = D;
-
-    // detect my id and mark the flag here
-    for (size_t s = 0; s < ctx->topology->numShards; s++) {
-        for (size_t n = 0; n < ctx->topology->shards[s].numNodes; n++) {
-            if (ctx->my_id && !strcmp(ctx->topology->shards[s].nodes[n].id, ctx->my_id)) {
-                ctx->topology->shards[s].nodes[n].flags |= MRNode_Self;
-            }
-        }
-    }
-
-err:
-   if (ctx->ok == 0) {
-        MRClusterTopology_Free(D);
-   }
-}
-
-cluster ::= cluster MYID shardid(B) . {
-    ctx->my_id = B;
-}
-
-cluster ::= cluster HASHFUNC STRING(A) NUMSLOTS INTEGER(B) . {
-    ctx->shardFunc = A.strval;
-    ctx->numSlots = B.intval;
-}
-
-
-cluster ::= cluster HASREPLICATION . {
-    ctx->replication = 1;
-}
```

Comment:
Do you cover in the new parsing this option as well? Is it still valid...?

Reply:
1. I’m not sure if it was ever valid. The parser in LibMR doesn’t include it
2. This `replication` value was never used. 

@MeirShpilraien WDYT?